modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java

```diff
@@ -367,6 +368,23 @@ public int getOneToManyObjectEntriesCount(
 			groupId, objectRelationshipId, primaryKey, related, search);
 	}
 
+	@Override
+	public ObjectEntry getOrAddEmptyObjectEntry(
+			String externalReferenceCode, long groupId, long userId,
+			long objectDefinitionId)
+		throws PortalException {
+
+		if (!ObjectEntryThreadLocal.isSkipObjectEntryResourcePermission()) {
+			_checkAddObjectEntryPortletResourcePermission(
+				groupId, objectDefinitionId, 0L, new HashMap<>());
+		}
```

Comment:
I recall that we need to modify the `_checkAddObjectEntryPortletResourcePermission()` to bypass the Account Restricted Entries permission.